### PR TITLE
MGMT-13302: MGMT-13358: Fix multiple OLM - CPU architecture related bugs

### PR DESCRIPTION
**Pull request by eliorerz**
Fix multiple OLM related bugs:

- [MGMT-13302](https://issues.redhat.com//browse/MGMT-13302): Return a bad request if the user wants to install not supported operator for arm64 
- [MGMT-13358](https://issues.redhat.com//browse/MGMT-13358): Fix multi-arch image - validation failure for CNV operator

## List all the issues related to this PR

- [ ] New Feature 
- [ ] Enhancement 
- [x] Bug fix
- [ ] Tests
- [ ] Documentation
- [ ] CI/CD 

## What environments does this code impact?

- [ ] Automation (CI, tools, etc)
- [x] Cloud
- [ ] Operator Managed Deployments
- [ ] None

## How was this code tested?

- [ ] assisted-test-infra environment
- [ ] dev-scripts environment
- [ ] Reviewer's test appreciated
- [x] Waiting for CI to do a full test run
- [x] Manual (Elaborate on how it was tested)
- [ ] No tests needed

### CODE REVIEW

File: internal/operators/manager.go [hunk 1]
```diff
@@ -401,6 +403,29 @@ func (mgr *Manager) GetSupportedOperatorsByType(operatorType models.OperatorType
...
+		monitoredOperators = append(monitoredOperators, monitoredOperator.Name)
+	}
+
+	for operatorName, operator := range mgr.olmOperators {
+		if !funk.ContainsString(monitoredOperators, operatorName) {
+			continue
+		}
+
+		if !operator.HasArmSupport() {
+			return errors.Errorf("Currently, you cannot install %s operator along side %s architecture.", operator.GetName(), cluster.CPUArchitecture)
```
Comment (gamli75) [inline internal/operators/manager.go:422]: @nirfarkas in the UI we have the following message: "OpenShift Virtualization is not available when ARM CPU architecture is selected" see: https://github.com/openshift-assisted/assisted-ui-lib/blob/4829bc07b9a69be382113ba60d407544aa73931e/src/ocm/components/featureSupportLevels/featureStateUtils.ts#L109 
Can you review the BE message?

File: internal/operators/manager.go [hunk 2]
```diff
@@ -448,3 +473,17 @@ func EnsureLVMAndCNVDoNotClash(cluster *common.Cluster, openshiftVersion string,
...
 	return nil
 }
+
+func (mgr *Manager) EnsureOperatorPrerequisite(cluster *common.Cluster, openshiftVersion string, operators []*models.MonitoredOperator) error {
+	err := EnsureLVMAndCNVDoNotClash(cluster, openshiftVersion, operators)
+	if err != nil {
+		return err
+	}
+
+	err = mgr.EnsureOperatorArchCapability(cluster, operators)
```
Comment (gamli75) [inline internal/operators/manager.go:483]: maybe we can skip this check if OCP version < 4.11, WDYT?

Comment (eliorerz) [reply internal/operators/manager.go:483]: To my opinion this have the same affect and the ~same running time. 
Only if we want to make it be more explicit (but it will make the code more complex over time)

File: internal/operators/manager.go [hunk 3]
```diff
@@ -401,6 +403,29 @@ func (mgr *Manager) GetSupportedOperatorsByType(operatorType models.OperatorType
 	return operators
 }
 
+func (mgr *Manager) EnsureOperatorArchCapability(cluster *common.Cluster, operators []*models.MonitoredOperator) error {
+	var monitoredOperators []string
+	if cluster.CPUArchitecture == common.X86CPUArchitecture {
```
[the remark this answers is not in the document]
Comment (gamli75) [reply internal/operators/manager.go:408]: cluster.CPUArchitecture == common.MultiCPUArchitecture is not enough - you need to check also the infra-env
infra-env.cpu_architecture=x86_64
see: https://issues.redhat.com/browse/MGMT-13358

[the remark this answers is not in the document]
Comment (gamli75) [reply internal/operators/manager.go:408]: regarding the generic suggestion GetSupportedArchitectures - I think it's better

File: internal/operators/manager.go [hunk 4]
```diff
@@ -401,6 +403,29 @@ func (mgr *Manager) GetSupportedOperatorsByType(operatorType models.OperatorType
...
+	for _, monitoredOperator := range operators {
+		monitoredOperators = append(monitoredOperators, monitoredOperator.Name)
+	}
+
+	for operatorName, operator := range mgr.olmOperators {
+		if !funk.ContainsString(monitoredOperators, operatorName) {
+			continue
+		}
+
+		if !operator.HasArmSupport() {
```
[the remark this answers is not in the document]
Comment (gamli75) [reply internal/operators/manager.go:421]: please add a test for this case

File: internal/operators/manager.go (hunk 3 above)
[the remark this answers is not in the document]
Comment (eliorerz) [reply internal/operators/manager.go:408]: > cluster.CPUArchitecture == common.MultiCPUArchitecture is not enough - you need to check also the infra-env infra-env.cpu_architecture=x86_64 see: https://issues.redhat.com/browse/MGMT-13358

I'm not sure we have infra-env on that stage (register cluster)

File: internal/operators/manager.go (hunk 4 above)
[the remark this answers is not in the document]
Comment (eliorerz) [reply internal/operators/manager.go:421]: Already did, will push it with the other fixes

File: data/default_os_images.json [hunk 5]
```diff
@@ -80,7 +80,7 @@
     {
         "openshift_version": "4.13",
         "cpu_architecture": "arm64",
-        "url": "https://mirror.openshift.com/pub/openshift-v4/aarch64/dependencies/rhcos/4.12/4.12.0/rhcos-4.12.0-aarch64-live.aarch64.iso",
+        "url": "`https://mirror.openshift.com/pub/openshift-v4/aarch64/dependencies/rhcos/4.12/4.12.0/rhcos-4.12.0-aarch64-live.aarch64.iso`",
```
Comment (gamli75) [inline data/default_os_images.json:83]: why this change is related to this PR?

File: internal/operators/manager.go (hunk 1 above)
Comment (nirfarkas) [reply internal/operators/manager.go]: @gamli75 After discussing this with @eliorerz we've decided to align the bad request message with the validations message so both will be:
"<operator name> is not available when <CPU arcitecture> architecture is selected"

File: data/default_os_images.json (hunk 5 above)
Comment (eliorerz) [reply data/default_os_images.json:83]: Not related, probably key was pressed by mistake

Comment (eliorerz) [reply data/default_os_images.json]: Fixed

File: internal/operators/lvm/lvm_operator.go [hunk 6]
```diff
@@ -53,6 +53,10 @@ func (o *operator) GetName() string {
 	return Operator.Name
 }
 
+func (o *operator) GetFullName() string {
+	return "Logical Volume Management"
```
Comment (gamli75) [inline internal/operators/lvm/lvm_operator.go:57]:
```suggestion
	return "Logical Volume Manager"
```

**Review by gamli75 — APPROVED** [review]
